kubetest2/pkg/app/testers/standard/ginkgo/ginkgo.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ginkgo
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"k8s.io/test-infra/kubetest2/pkg/app/testers"
+	"k8s.io/test-infra/kubetest2/pkg/app/testers/standard/ginkgo/kubectl"
+	"k8s.io/test-infra/kubetest2/pkg/exec"
+	"k8s.io/test-infra/kubetest2/pkg/types"
+
+	"github.com/spf13/pflag"
+)
+
+const (
+	binary = "ginkgo" // TODO(RonWeber): Actually find these binaries.
+
+)
+
+var (
+	kubeRoot     string
+	platformPath string
+)
+
+const usage = "Eh, we'll figure it out."
+
+func init() {
+	testers.Register("ginkgo", usage, NewTester)
+}
+
+// Tester implements a kubetest2 types.Tester that exec's it's arguments
+type Tester struct {
+	conformanceTest bool
+
+	flakeAttempts string
+	parallel      string
+	host          string
+	skipRegex     string
+
+	deployer types.GinkgoableDeployer
+}
+
+// NewTester creates a new Tester
+func NewTester(common types.Options, testArgs []string, deployer types.Deployer) (types.Tester, error) {
+	wd, _ := os.Getwd()
+	kubeRoot = filepath.Join(wd, "kubernetes")                            //TODO(RonWeber): Real logic to get kubernetes
+	platformPath = filepath.Join(kubeRoot, "platforms", "linux", "amd64") //TODO(RonWeber): Real logic to figure out current platform.
+
+	t := Tester{}
+
+	gd, ok := deployer.(types.GinkgoableDeployer)
+	if !ok {
+		return nil, fmt.Errorf("the ginkgo tester does not support this deployer")
+	}
+	t.deployer = gd // We'll need to do stuff with the deployer after cluster is up, but before tests.
+
+	flags := bindFlags(&t)
+	flags.Parse(testArgs)
+
+	t.conformanceTest = true //TODO(RonWeber): Better logic here.
+
+	return t, nil
+}
+
+func bindFlags(t *Tester) *pflag.FlagSet {
+	flags := pflag.NewFlagSet("ginkgo tester", pflag.ContinueOnError)
+	flags.StringVar(&t.flakeAttempts, "flake-attempts", "1", "Make up to this many attempts to run each spec.")
+	flags.StringVar(&t.parallel, "parallel", "1", "Run this many tests in parallel at once.  Defaults to 1 (Not parallel)")
+	flags.StringVar(&t.skipRegex, "skip", "", "Regular expression of jobs to skip.") //TODO(RonWeber): Some of that can be detected automatically.
+	return flags
+}
+
+// Test runs the test
+func (t Tester) Test() error {
+	if err := t.pretestSetup(); err != nil {
+		return err
+	}
+
+	provider := t.deployer.Provider()
+	if t.conformanceTest {
+		provider = "skeleton"
```

Comment:
we should probably allow the user to override this with a flag later

Reply:
Separate from the (eventual) conformanceTest flag?

Reply:
yeah, conformance test might even need to be another tester

Reply:
Leaving this for the future.